src/test/app/PermissionedDomains_test.cpp

```diff
@@ -0,0 +1,478 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2024 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <xrpld/app/tx/detail/PermissionedDomainSet.h>
+#include <xrpld/ledger/ApplyViewImpl.h>
+#include <xrpl/basics/Blob.h>
+#include <xrpl/basics/Slice.h>
+#include <xrpl/protocol/Feature.h>
+#include <xrpl/protocol/Issue.h>
+#include <xrpl/protocol/jss.h>
+#include <algorithm>
+#include <map>
+#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace ripple {
+namespace test {
+namespace jtx {
+
+class PermissionedDomains_test : public beast::unit_test::suite
+{
+    FeatureBitset withFeature_{
+        supported_amendments() | featurePermissionedDomains};
+    FeatureBitset withoutFeature_{supported_amendments()};
+
+    using Credential = std::pair<AccountID, Blob>;
+    using Credentials = std::vector<Credential>;
+
+    // helpers
+    // Make json for PermissionedDomainSet transaction
+    static Json::Value
+    setTx(
```

Comment:
Maybe put these functions (setTx + deleteTx) into dedicated file as they will be reused by future code. Like DepositAuth_test.cpp / deposit.cpp

Reply:
fixed